src/components/Project/ProjectBasics/ProjectForm/ProjectHashTags/ProjectHashTags.tsx

```diff
@@ -149,24 +173,65 @@ const ProjectHashTagsDialog: FC<IProjectHashTagsDialogProps> = forwardRef(
       [hashTagsObject, hashTagsForSubmit],
     );
 
+    const handleConcat = (hashtagsForSubmit: IListItem[]) => {
+      
+      // Combine the hashtags that aren't yet in hashTagsForSubmit with the ones that are there
+      if (hashTagsForSubmit.length) {
+        const existingHashtagIds = new Set(hashTagsForSubmit.map((hashtag) => hashtag.id));
+        // Filter out the hashtags that already exist in hashTagsForSubmit and after that combine the new values with the ones in hashTagsForSubmit
+        const newHashtags = hashtagsForSubmit.filter(hashtag => !existingHashtagIds.has(hashtag.id));
+        const combinedHashtags = [...hashTagsForSubmit, ...newHashtags];
+        // when user searches for hashtags and adds one of the search results, the hashtag will come twice --> uniqWith deletes all duplicate objects
+        return _.uniqWith(combinedHashtags, _.isEqual);
+      } 
+      return _.uniqWith(hashTagsForSubmit, _.isEqual);
+    }
+
+    useEffect(() => {
+      if (projectMode === 'new' && hashTagsForSubmit) {
+        const hashtags = handleConcat(hashTagsForSubmit);
+        setHashTagsState((current) => ({
+          ...current,
+          projectHashTags: hashtags
+        }));
+      }
+      setHashtagClicked(false);
+    },[hashtagClicked])
```

Comment:
Could the state be modified only when the hashTagDialog is closed by clicking "save", or what is the need for modifying it after every hashTagClick?

Reply:
This must've also be connected to the older logic since now it's not needed anymore